frotz.py

```diff
@@ -126,15 +131,45 @@ def start_game(self, game_file):
             # print a welcome banner and pause for a moment
             # that way the end user will have a chance to see which version of frotz we are using
             # and which file we are loading
-
-            if platform.machine().startswith('arm'):
-                logging.error('ARM not supported, yet') # FIXME
-                sys.exit(0)
+            if platform.version().find("Ubuntu") > -1 or platform.version().find("Debian") > -1:
+                cmd = '/usr/games/frotz'
             else:
-                if platform.architecture()[0] == '64bit':
-                    self._vte.feed_child("cd '%s'; clear; frotz64|head -3 ; echo '\nLoading %s...'; sleep 2; frotz64 '%s'; exit\n" % (save_dir, os.path.basename(game_file), game_file))
-                else:
-                    self._vte.feed_child("cd '%s'; clear; frotz32|head -3 ; echo '\nLoading %s...'; sleep 2; frotz32 '%s'; exit\n" % (save_dir, os.path.basename(game_file), game_file))
+                cmd = '/usr/bin/frotz'
+            if os.path.isfile(cmd) and os.access(cmd, os.X_OK):
+                logging.debug('Using frotz installed in the system')
+                self._vte.feed_child(
+                    (
+                            "cd '%s'; "
+                            "clear; "
+                            "frotz|head -3 ; "
+                            "echo '\nLoading %s...'; "
+                            "sleep 2; frotz '%s'; "
+                            "exit\n" % (
+                                save_dir,
+                                os.path.basename(game_file),
+                                game_file)).encode('utf-8')
+                )
+            else:
+                dialog = Gtk.MessageDialog(
+                    transient_for=self,
+                    flags=0,
+                    message_type=Gtk.MessageType.INFO,
+                    buttons=Gtk.ButtonsType.OK,
+                    text="frotz is not installed"
+                )
+                dialog.format_secondary_text(
+                    "Install frotz by clicking OK and then restart the activity"
+                )
+                response = dialog.run()
+
+                if response == Gtk.ResponseType.OK:
+                    logging.debug("Installing frotz")
+                    if platform.version().find("Ubuntu") > -1 or platform.version().find("Debian") > -1:
+                        cmd = "apt install frotz"
```

Comment:
Doesn't the user have to be root in order to use these commands?  We don't run Sugar activities as root.